chia/wallet/trading/trade_store.py

```diff
@@ -236,6 +301,25 @@ async def get_trade_record_with_status(self, status: TradeStatus) -> List[TradeR
 
         return records
 
+    async def get_coins_of_interest_with_trade_statuses(self, trade_statuses: List[TradeStatus]) -> List[Coin]:
+        """
+        Checks DB for TradeRecord with id: id and returns it.
+        """
+        async with self.db_wrapper.reader_no_transaction() as conn:
+            cursor = await conn.execute(
```

Comment:
if you use `execute_fetchall()` here instead, you can avoid the extra two round-trips to the sqlite thread in the `fetchall()` and `close()` calls